src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -199,7 +199,7 @@ public static bool TryParse(string value, bool allowFloating, out VersionRange v
             if (!string.IsNullOrWhiteSpace(minVersionString))
             {
                 // parse the min version string
-                if (allowFloating && minVersionString.Contains("*"))
+                if (allowFloating && minVersionString.Contains('*'))
```

Comment:
This fixes https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1847 analyzer warning.

Reply:
considering we use treat warnings as errors, does this mean you forgot to remove a supression?  Otherwise I don't understand why this line would start reporting ca1847 in this branch.

Reply:
We have not yet enabled `CA1847` in the repo. I am planning on spending sometime this sprint to re-review analyzers turned on the repo and propose new rules and turn them on after approvals. I remember you also asked to check why [`CA1068: CancellationToken parameters must come last` ](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1068) is not turned on currently.